packages/lesswrong/components/posts/PostsPage/PostsPageSplashHeader.tsx

```diff
@@ -0,0 +1,572 @@
+  const backgroundImgCropPreviewRef = useRef<HTMLDivElement>(null);
+
+  const transitionHeader = (headerVisibile: boolean) => {
+    setToCVisible(!headerVisibile);
+    setVisible(headerVisibile);
+  };
+
+  const observerRef = useObserver<HTMLDivElement>({
+    onEnter: () => transitionHeader(true),
+    onExit: () => transitionHeader(false),
+    threshold: 0.95
+  });
+
+  const setCropPreview = (coordinates?: Coordinates) => {
+    if (imgRef.current && backgroundImgWrapperRef.current && backgroundImgCropPreviewRef.current) {
+      if (coordinates) {
+        const updatedMask = `
+          linear-gradient(#000 0 0) ${coordinates.x}px ${coordinates.y}px/${coordinates.width}px ${coordinates.height}px,
+          linear-gradient(rgba(0,0,0,0.4) 0 0)
+        `;
+        imgRef.current.style.mask = `${updatedMask} no-repeat`;
+        imgRef.current.style.webkitMask = updatedMask;
+        imgRef.current.style.webkitMaskRepeat = 'no-repeat';
+
+        setCropPreviewEnabled(true);
+      } else {
+        imgRef.current.style.mask = '';
+        imgRef.current.style.webkitMask = '';
+        imgRef.current.style.webkitMaskRepeat = '';
+
+        setCropPreviewEnabled(false);
+      }
+    }
+  };
+  
+  const { results: reviews } = useMulti({
+    terms: {
+      view: "reviews",
+      postId: post._id,
+    },
+    collectionName: "Comments",
+    fragmentName: 'CommentWithRepliesFragment',
+    fetchPolicy: 'cache-and-network',
+    limit: 5
+  });
+
+  useEffect(() => {
+    const postLastSavedImage = post.reviewWinner.reviewWinnerArt?.splashArtImageUrl;
+    const newBackgroundImage = selectedImageInfo?.splashArtImageUrl || postLastSavedImage;
+
+    if (newBackgroundImage) {
+      setBackgroundImage(newBackgroundImage.replace('upload/', imageFlipped ? 'upload/a_hflip/' : 'upload/'));
+    }
+  }, [post, selectedImageInfo, imageFlipped]);
+
+  const nonhumanAudio = post.podcastEpisodeId === null && isLWorAF;
+
+  const audioIcon = (
+    <LWTooltip title={'Listen to this post'} className={classNames(classes.togglePodcastContainer, {[classes.nonhumanAudio]: nonhumanAudio, [classes.audioIconOn]: showEmbeddedPlayer})}>
+      <a href="#" onClick={toggleEmbeddedPlayer}>
+        <ForumIcon icon="VolumeUp" className={classNames(classes.audioIcon, {})} />
+      </a>
+    </LWTooltip>
+  );
+
+  const votingSystem = getVotingSystemByName(post.votingSystem ?? 'default');
+  const postActionsButton = <PostActionsButton post={post} className={classes.postActionsButton} flip />;
+
+  const topLeftSection = (
+    <div className={classes.leftSection}>
+      <Link className={classes.reviewNavigation} to="/bestoflesswrong?sort=year">
+        Ranked #{post.reviewWinner.reviewRanking + 1} of {post.reviewWinner.competitorCount} posts in the {post.reviewWinner.reviewYear} Review
+      </Link>
+      <Link className={classes.reviewNavigationMobile} to="/bestoflesswrong?sort=year">
+        #{post.reviewWinner.reviewRanking + 1} in 2021 Review
```

Comment:
I think the year here needs to come from the `reviewWinner` as well?

Reply:
Done